library_generation/requirements.in

```diff
@@ -0,0 +1,27 @@
+# add new packages here and run `pip-compile requirements.in --generate-hashes` to
+# generate an entry with hashes in requirements.txt
+absl-py==2.1.0
+attr==0.3.2
+attrs==23.2.0
+black==24.4.2
+click==8.1.7
+gitdb==4.0.11
+GitPython==3.1.43
+Jinja2==3.1.4
+lxml==5.2.2
+MarkupSafe==2.1.5
+mypy-extensions==1.0.0
+packaging==23.2
+pathspec==0.12.1
+PyYAML==6.0.1
+smmap==5.0.1
+typing==3.7.4.3
+parameterized==0.9.0 # used in parameterized test
+colorlog==6.8.2
+watchdog==4.0.1
+nox==2024.4.15
+requests==2.32.3
+setuptools==65.5.1
+jinja2==3.1.4
+deprecation==2.1.0
+protobuf==3.20.2
```

Comment:
Did you find which functions use `deprecation` and `protobuf` package?

Reply:
I didn't look at the functions but tried running the generation without these. It worked, so I'm removing them. Thanks for catching this.